cognite/powerops/preprocessor/cogreader.py

```diff
@@ -103,6 +147,22 @@ def get_extra_files_metadata(self) -> List[dict[str, Union[str, int]]]:
         else:
             return []
 
+    def get_cog_shop_file_list(self) -> list[dict]:
+        cog_shop_files = []
+
+        for file in self.cog_shop_file_loader.file_load_sequence:
+            if file.external_id and file.pick == "latest":
+                cog_shop_files.append({"external_id": file.external_id, "file_type": file.file_type})
+            elif file.external_id_prefix and file.pick == "closest":
+                files = self.client.files.list(external_id_prefix=file.external_id_prefix, limit=None)
+                if closest_file := find_closest_file(files, self.cog_shop_config.starttime_ms):
+                    cog_shop_files.append({"external_id": closest_file.external_id, "file_type": file.file_type})
+                else:
+                    raise CogReaderError(
+                        f"Couldn't find a valid cut file for start time " f"{self.cog_shop_config.starttime_ms}"
+                    )
```

Comment:
There seems to be a few cases which are not captured here, for example, `file.pick = "all"`. Suggest adding an else clause raising an appropriate error. Or log that you failed to read some file and continue.

Reply:
I removed it. Doesn't make sense to have that filtering option it at the moment